ocrd_cis/ocropy/binarize.py

```diff
@@ -105,146 +92,122 @@ def process(self):
-                zoom = 300.0/dpi
-            else:
-                zoom = 1
-            
-            if level == 'page':
-                self.process_page(page, page_image, page_xywh, zoom,
-                                  input_file.pageId, file_id)
-            else:
-                if level == 'table':
-                    regions = page.get_TableRegion()
-                else: # region
-                    regions = page.get_AllRegions(classes=['Text'], order='reading-order')
-                if not regions:
-                    self.logger.warning('Page "%s" contains no text regions', page_id)
-                for region in regions:
-                    region_image, region_xywh = self.workspace.image_from_segment(
-                        region, page_image, page_xywh, feature_filter='binarized')
-                    if level == 'region':
-                        self.process_region(region, region_image, region_xywh, zoom,
-                                            input_file.pageId, file_id + '_' + region.id)
-                        continue
-                    lines = region.get_TextLine()
-                    if not lines:
-                        self.logger.warning('Page "%s" region "%s" contains no text lines',
-                                            page_id, region.id)
-                    for line in lines:
-                        line_image, line_xywh = self.workspace.image_from_segment(
-                            line, region_image, region_xywh, feature_filter='binarized')
-                        self.process_line(line, line_image, line_xywh, zoom,
-                                          input_file.pageId, region.id,
-                                          file_id + '_' + region.id + '_' + line.id)
+        pcgts = input_pcgts[0]
+        page = pcgts.get_Page()
+        assert page
 
-            # update METS (add the PAGE file):
-            file_path = os.path.join(self.output_file_grp, file_id + '.xml')
-            pcgts.set_pcGtsId(file_id)
-            out = self.workspace.add_file(
-                ID=file_id,
-                file_grp=self.output_file_grp,
-                pageId=input_file.pageId,
-                local_filename=file_path,
-                mimetype=MIMETYPE_PAGE,
-                content=to_xml(pcgts))
-            self.logger.info('created file ID: %s, file_grp: %s, path: %s',
-                             file_id, self.output_file_grp, out.local_filename)
+        page_image, page_xywh, page_image_info = self.workspace.image_from_page(page, page_id, feature_filter='binarized')
+        zoom = determine_zoom(self.parameter['dpi'], page_image_info)
+        self.logger.info(f"Page '{page_id}' uses {self.parameter['dpi']} DPI")
+        
+        ret = [pcgts]
+        if level == 'page':
+            try:
+                ret.append(self.process_page(page, page_image, page_xywh, zoom, page_id, output_file_id))
+            except ValueError as e:
```

Comment:
@bertsky Do we even want to catch these or should we let them explode and let core do the error handling? For page-wide binarization, I think we probably should because that means the page failed. But for region and line, we might have rogue instances of zero size but all the other regions/lines might be fine.